fast_ms/src/diff/compress_ms.cpp

```diff
@@ -0,0 +1,913 @@
+            n_zeros++;
+            c_ms[out_idx++] = 0;
+            continue;
+        }
+
+        size_type new_one_enc = compute_length(n_ones, n_zeros);
+        for(size_type k=0; k<new_one_enc; k++)
+            c_ms[out_idx++] = 1;
+        n_ones = 0;
+        c_ms[out_idx++] = 0;
+        n_zeros = 1;
+    }
+    if (n_ones > 0) {
+        size_type new_one_enc = compute_length(n_ones, n_zeros);
+        for(size_type k=0; k<new_one_enc; k++)
+            c_ms[out_idx++] = 1;
+    }
+    return diff_from(from);
+}
+
+
+template<typename vec_type, typename enc_type>
+size_type comp1(const string ms_path, const InputFlags& flags, const string outputDirectory, const string tablesDirectory, const uint32_t threshold, const size_type nZeros, const size_type nOnes, uint8_t allowNegativeMS, uint8_t greedyStrategy, uint8_t verbose) {
+	uint32_t initialMSValue;
+	sdsl::bit_vector ms;
+	
+cerr << "WTF> 1 \n";	
+	
+	sdsl::load_from_file(ms,ms_path);
+cerr << "WTF> 2 \n";
+	sdsl::bit_vector permutedMS(ms.size(),0);
+cerr << "WTF> 3 \n";
+	
+	
+	// Permuting
+	loadTables(tablesDirectory,threshold,nZeros,nOnes,allowNegativeMS);
+	permuteBitvector(ms,permutedMS,threshold,allowNegativeMS,greedyStrategy,verbose);
+	cerr << "Permuting completed" << endl;
+	for (initialMSValue=0; initialMSValue<threshold; initialMSValue++) free(neighbor[initialMSValue]);
+	free(neighbor);
+	equalMSValues(ms,permutedMS,threshold,verbose);
+
+
+    size_type from = abs_point();
+    enc_type encoder(32);
+    histo_t counter;
+    size_type n_runs = fill_encoder<enc_type>(permutedMS, encoder, counter);
+    vec_type c_ms(encoder, permutedMS.size());	
+	char buffer[100]; 
+	sprintf(buffer,"%s/compressedMS-diff-%d-%d-%d",outputDirectory.c_str(),threshold,allowNegativeMS,greedyStrategy);
+    std::ofstream out{  buffer   /*ms_path + ms_compression::to_str(flags.compression)*/, std::ios::binary};
+    c_ms.writeTo(out);
+
+    (cerr << n_runs << " runs over "
+          << c_ms.getSize() << " elements ("
+          << c_ms.getSize() / static_cast<float>(n_runs) << " elements / run)"
+          << endl);
+    //for(auto item : counter)
+    //    cout << item.first << "," << item.second << endl;
+    return diff_from(from);
+}
+
+
+int main(int argc, char **argv){
+    OptParser input(argc, argv);
+    string ms_path;
+
+    string help__ms_path{"\t-ms_path <path to a binary>: the ms bit vector\n"};
+    if(argc == 1){
+        (cerr << "Compress a ms vector. Creates files <ms_path>.xxx\n"
+              << "Args:\n"
+              << help__ms_path
+              << help__compression
+              << endl);
+        exit(0);
+    }
+    InputFlags flags;
+    try{
+        flags = InputFlags(input);
+    }
+    catch (string s) {
+        cerr << s << endl;
+        return 1;
+    }
+    ms_path = input.getCmdOption("-ms_path");
+    cerr << "check: " << diff_from(abs_point()) << endl;//size_type mem_mark = abs_point();
+    switch(flags.compression)
```

Comment:
this program works only for `compression=rle` right? 

in that case we should throw an error for all other cases.

Reply:
yes, it works only for compression=rle. rather than throwing an error for all other cases, i would just remove the compression= flag and add RLE to the short description/documentation of the program.